File: Fusion/CreateDomain/CreateDomain.py
# -*- coding: utf-8 -*-
from qiniu import Auth, QiniuMacAuth
import requests, json, sys
  
access_key = 'ak'
secret_key = 'sk'
  
auth = Auth(access_key, secret_key)
  
url = 'http://api.qiniu.com/domain/just-test-001.qiniuts.com'
body = {
    "type": "normal",
    "platform": "web",
    "geoCover": "china",
    "protocol": "http",
    "source": {
        "sourceType": "qiniuBucket",
        "SourceQiniuBucket": "aaaaa"
    },
    "cacheControls": [
        {
            "time": 1,
            "timeunit": 5,
            "type": "all",
            "rule": "*"
        }
    ],
    "ignoreParam": False
}
  
token = 'QBox ' + auth.token_of_request(url, json.dumps(body), 'application/json')
print token
  
r = requests.post(url, data=json.dumps(body), headers={'Authorization': token, 'Content-Type': 'application/json'})
  
print r
print r.text

File: Fusion/CreateDomain/CreateDomain.php
<?php
require '<srcfolderpath>/autoload.php';
use Qiniu\Auth;
use Qiniu\Http\Client;
 
$accessKey = 'ak';
$secretKey = 'sk';
$auth = new Auth($accessKey, $secretKey);
 
$domain='just-test-003.qiniuts.com';
$data=array(
    'type' => 'normal',
    'platform' => 'web',
    'geoCover' => 'china',
    'protocol' => 'http',
    'source' => array (
        'sourceType' => 'qiniuBucket',
        'SourceQiniuBucket' => 'aaaaa'
    ),
    'cacheControls' => array (
        array(
            'time' => 1,
            'timeunit' => 5,
            'type' => 'all',
            'rule' => '*'
        )
    ),
    'ignoreParam' => FALSE
);
//echo json_encode($data)."\n";
 
$url='http://api.qiniu.com/domain/'.$domain;
$token=$auth->authorization($url, json_encode($data), 'application/json')['Authorization'];
echo $token."\n";
 
$client = new Client();
$headers=array(
    'Authorization' => $token,
    'Content-Type' => 'application/json'
);
$res=$client->post($url, json_encode($data), $headers);
echo $res->statusCode."\n";
echo print_r($res->headers)."\n";
echo $res->body."\n";
?>
